### PR TITLE
Fix resources handling in admin

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR uniformize and mutualize resources and community resources handling in admin.

Modals become routables and totaly mutualized.

### CODE REVIEW

**Comment by vinyll** [comment]
My feedbacks end up to be only _ES6_ related and are questionable.

File: js/components/dataset/resource/file-form.vue [hunk 1]
```diff
@@ -110,14 +111,17 @@ export default {
...
+        },
+        upload_endpoint() {
             let operations = API.datasets.operations;
             let params = {};
             let endpoint = 'upload_';
             if (typeof this.dataset !== 'undefined') {
                 params = {dataset: this.dataset.id};
             }
-            if (this.community) {
+            if (this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/file-form.vue]: Does it work without parenthesis?

File: js/components/dataset/resource/modal.vue [hunk 2]
```diff
@@ -99,29 +104,100 @@
...
             confirm: false,
-            dataset: {}
+            dataset: new Dataset(),
+            resource: new Resource(),
+            next_route: null
         };
     },
+    computed: {
+        can_edit() {
+            if (this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: Same here

File: js/components/dataset/resource/modal.vue [hunk 3]
```diff
@@ -99,29 +104,100 @@
...
+                return this.$root.me.can_edit(this.resource);
+            } else {
+                return this.$root.me.can_edit(this.dataset);
+            }
+        },
+        is_community() {
+            return this.resource instanceof CommunityResource
```
Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: `;`

File: js/components/dataset/resource/modal.vue [hunk 4]
```diff
@@ -99,29 +104,100 @@
...
+                this.resource = new CommunityResource();
+                this.resource.fetch(this.$route.params.rid);
+            }
+        }
+    },
     methods: {
         save: function() {
             if (this.$refs.form.validate()) {
-                this.dataset.save_resource(this.$refs.form.serialize());
+                if (this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: Same here

File: js/components/dataset/resource/modal.vue [hunk 5]
```diff
@@ -99,29 +104,100 @@
...
+                    this.dataset.save_resource(this.$refs.form.serialize());
+                }
                 this.$refs.modal.close();
                 return true;
             }
         },
         delete_confirmed: function() {
-            this.dataset.delete_resource(this.resource.id);
-            this.$refs.modal.close();
+            if (this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: Same here

File: js/components/dataset/resource/modal.vue [hunk 6]
```diff
@@ -99,29 +104,100 @@
...
+                );
+            } else {
+                this.dataset.delete_resource(this.resource.id);
+                this.$refs.modal.close();
+            }
+        }
+    },
+    watch: {
+        'dataset.resources': function(resources) {
+            if (!this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: Same here

File: js/components/dataset/resource/file-form.vue [hunk 7]
```diff
@@ -140,7 +144,7 @@ export default {
             this.progress = Math.round(uploaded * 100 / total);
         },
         'uploader:complete': function(id, response) {
-            if (!this.community) {
+            if (!this.is_community) {
```
Comment (davidbgk) [inline js/components/dataset/resource/file-form.vue]: Same here

File: udata/templates/dataset/resource/list-item.html [hunk 8]
```diff
@@ -38,13 +38,13 @@ <h4 class="list-group-item-heading ellipsis">
...
-            {% set edit_url = url_for('admin.index', path='dataset/{id}/community-resource/{rid}'.format(id=resource.dataset.id, rid=resource.id)) %}
+            {% set edit_path = 'dataset/{id}/community-resource/{rid}' %}
         {% else %}
-            {% set edit_url = url_for('admin.index', path='dataset/{id}/'.format(id=dataset.id), **{'resource_id': resource.id}) %}
+            {% set edit_path = 'dataset/{id}/resource/{rid}' %}
         {% endif %}
         <div class="btn-group btn-group-xs tools">
             <a class="btn btn-default" rel="tooltip" title="{{ _('Edit') }}"
-                href="{{ edit_url }}">
+                href="{{ url_for('admin.index', path=edit_path.format(id=dataset.id, rid=resource.id)) }}">
```
Comment (davidbgk) [inline udata/templates/dataset/resource/list-item.html]: Not part of that PR but a `url_for_admin` method would be handy. Thoughts?

File: js/components/dataset/resource/modal.vue [hunk 9]
```diff
@@ -99,29 +104,100 @@
...
     },
+    computed: {
+        can_edit() {
+            if (this.is_community) {
+                return this.$root.me.can_edit(this.resource);
+            } else {
+                return this.$root.me.can_edit(this.dataset);
+            }
+        },
+        is_community() {
```
Comment (vinyll) [inline js/components/dataset/resource/modal.vue]: To complete @davidbgk notice about this:  `get is_community() {` should solve.

File: js/components/dataset/resource/file-form.vue (hunk 1 above)
Comment (noirbizarre) [inline js/components/dataset/resource/file-form.vue]: Yep, this is a computed property, meant to be called without parenthesis.

File: udata/templates/dataset/resource/list-item.html (hunk 8 above)
Comment (noirbizarre) [inline udata/templates/dataset/resource/list-item.html]: Agreed. I will submit another PR jsut for this helper.

File: js/components/dataset/resource/modal.vue (hunk 9 above)
Comment (noirbizarre) [inline js/components/dataset/resource/modal.vue]: Yep, but in Vue.js computed properties are reactive getters.

Comment (davidbgk) [inline js/components/dataset/resource/modal.vue]: Awesomely magical ;)

Comment (vinyll) [inline js/components/dataset/resource/modal.vue]: whouuuuu… implicit magic! :trollface:

**Comment by davidbgk** [comment]
r+wc

File: js/components/dataset/resource/modal.vue (hunk 9 above)
Comment (noirbizarre) [inline js/components/dataset/resource/modal.vue]: Not so implicit because it's in the computed object, but yep, quite magic ! :)

**Comment by noirbizarre** [comment]
@vinyll Comments on commit are quite hard to read :/

**Pull request merged**